### DISCUSSION TITLE
How do I display the full path instead of a shortened version?

### DISCUSSION THREAD

**Discussion by kontroll** [zimfw/zimfw#500] [General]
As it stands, I'm looking at a path that reads `/r/m/k/A/D/Etc` and I cannot for the life of me figure out what option to set for this to simply display `/run/media/kontroll/Another/Directory/Etc`.

Ideally it would be a matter of truncating directories in the path if there isn't room to display it all in one line (give or take, of course), but in this instance the terminal is 150 characters wide; there is beyond sufficient space to very comfortably display the full path.

Here's my zimrc, just in case:
```
zmodule eriner
zmodule zsh-users/zsh-completions --fpath src
zmodule completion
zmodule zsh-users/zsh-syntax-highlighting
zmodule zsh-users/zsh-autosuggestions
zmodule input
zmodule termtitle
zstyle ':zim:termtitle' hooks 'preexec' 'precmd'
zstyle ':zim:termtitle:preexec' format '${${(A)=1}[1]}'
zstyle ':zim:termtitle:precmd'  format '%1~'
zmodule duration-info
zmodule git-info
```

I've looked all over the place, and turned up blank. The closest thing I could find was the `prompt-pwd` module, but that didn't appear to have any effect whatsoever. Specifically using `zstyle ':zim:prompt-pwd:fish-style' dir-length 0` (I tried 20 instead of 0 as well), with or without a `zmodule prompt-pwd` thrown in for good measure (even though the module documentation does not mention this at all).

Any help would be appreciated.

**Comment by antoineco** [comment]
This behavior is hardcoded in the `eriner` prompt (first line of your config): [eriner.zsh-theme#L60-L62](https://github.com/zimfw/eriner/blob/7e90f5688c03755e334e9117bda71d8dff7d6ebe/eriner.zsh-theme#L60-L62).

Probably the quickest way to solve this is to fork the repository on GitHub, remove the 3 highlighted lines, then perform the following change to your config:

```diff
-zmodule eriner
+zmodule kontroll/eriner
```

Apply with `zimfw uninstall && zimfw install`.

**Reply by kontroll** [reply]
Applying turned out to be slightly more involved. First I had to remove eriner from the zimrc for the uninstall command to pick up that the module was superfluous. At first I simply deleted the dir manually and ran the install command, but I figured I'd try and find out why uninstall didn't appear to do anything. :)

Either way, it's working as desired now, which is wonderful.

Thank you!

**Comment by ericbn** [comment]
As a side note: Don't add `zstyle` calls to your .zimrc file. The .zimrc file is not sourced during startup and only used by the zimfw plugin manager to install the modules and create the static script that initializes them.

**Reply by kontroll** [reply]
I'll have to go back and look more closely at the documentation then. :)

**Comment by ericbn** [comment]
I've just updated the [eriner](https://github.com/zimfw/eriner) theme to require the prompt-pwd module. All the [prompt-pwd module settings](https://github.com/zimfw/prompt-pwd/blob/master/README.md#settings) now also apply to this theme.

This means you can add the following to the **end** of your ~/.zshrc file to override how the current directory is shown:
```
zstyle ':zim:prompt-pwd:fish-style' dir-length 0
```
Adding this line to your ~/.zshrc file after the `source ${ZIM_HOME}/init.zsh` line overrides what was set by the prompt theme.
